### ISSUE TITLE
False negative for invalid abstractmethod implementation

### ISSUE THREAD

**Issue by aatle** [python/mypy#17710]
**Bug Report**
If a subclass of an ABC implements an `abstractmethod` as an attribute (e.g. not as method or class variable), (and the subclass is instantiated,) then mypy passes. But it will always fail at runtime (as `ABCMeta` can only detect variables/methods on the class).
There are only some cases where this can happen because implementing a method as an attribute would yield a mypy type error most of the time:
- an abstract property is implemented as a normal instance attribute
- an abstract method is implemented as a function instance attribute

Note: these implementations are valid both pragmatically and for type-checking, but `ABCMeta` is obligated to raise an error because it cannot detect the implementation.

**To Reproduce**
The following program will pass in mypy but fail at runtime.
```python
from abc import abstractmethod, ABC
from collections.abc import Callable


class Method(ABC):
    @abstractmethod
    def f(self) -> None:
        ...

class ImplMethod(Method):
    f: Callable[[], None]
    def __init__(self):
        self.f = print

ImplMethod()


class Property(ABC):
    @property
    @abstractmethod
    def x(self) -> int:
        ...

class ImplProperty(Property):
    x: int
    def __init__(self):
        self.x = 2

ImplProperty()
```

**Expected Behavior**
Error at runtime.
Mypy should error too because the implementations do not satisfy `abstractmethod` at runtime.

**Actual Behavior**
Error at runtime.
Mypy success.

**Your Environment**
- Mypy version used: latest (1.11.1), with default playground flags
- Python version used: 3.12.4

Also:
It seems like mypy doesn't complain when using `abstractmethod` on a non-ABC class. This is intended, right?
